### PR TITLE
Fix panic when using a macros 1.1 custom derive on a struct containing a macro invocation

### LINKED ISSUE #38706: proc_macro does not support type macros
```rust
#![feature(proc_macro, proc_macro_lib)]

extern crate proc_macro;
use proc_macro::TokenStream;

#[proc_macro_derive(Nothing)]
pub fn nothing(_: TokenStream) -> TokenStream {
    "".parse().unwrap()
}
```

```rust
#[macro_use]
extern crate nothing_derive;

macro_rules! int {
    () => { i32 }
}

#[derive(Nothing)]
struct S {
    x: int!(),
}
```

```
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

thread 'rustc' panicked at 'visit_mac disabled by default', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libsyntax/visit.rs:102
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

### PR DESCRIPTION
**Pull request by keeperofdakeys**
Fixes #38706 

r? @jseyfried

### CODE REVIEW

File: src/librustc_resolve/macros.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ use syntax::ext::hygiene::Mark;
 use syntax::ext::tt::macro_rules;
 use syntax::feature_gate::{emit_feature_err, GateIssue};
 use syntax::fold::Folder;
+use syntax::fold;
```
Comment (jseyfried) [inline src/librustc_resolve/macros.rs]: nit: `use syntax::fold::{self, Folder}`.

File: src/libsyntax_ext/deriving/custom.rs [hunk 2]
```diff
@@ -28,6 +28,9 @@ impl<'a> Visitor<'a> for MarkAttrs<'a> {
             mark_known(attr);
         }
     }
+
+    fn visit_mac(&mut self, _mac: &Mac) {
+    }
```
Comment (jseyfried) [inline src/libsyntax_ext/deriving/custom.rs]: nit: I believe using a single line for empty functions (i.e. `fn f(...) {}`) is more idiomatic, at least in the compiler.

File: src/test/run-pass-fulldeps/proc-macro/struct-field-macro.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    () => { i32 }
+}
+
+#[derive(Nothing)]
+struct S {
+    x: int!(),
+}
+
+fn main() {
+}
```
Comment (jseyfried) [inline src/test/run-pass-fulldeps/proc-macro/struct-field-macro.rs]: (same here)

**Review by jseyfried — APPROVED** [review]
Thanks! r=me module nits

**Comment by dtolnay** [comment]
Thanks for the quick fix! I asked in https://github.com/rust-lang/rust/issues/35900#issuecomment-269857068 about backporting this to beta. Is there anything I need to do to make that happen?

**Comment by alexcrichton** [comment]
@dtolnay this is now tagged `beta-nominated` (thanks @jseyfried!) so it'll make its way through the process

**Comment by nikomatsakis** [comment]
Approving for beta. Small patch, regression. 

cc @rust-lang/compiler

**Pull request merged**